contributors/design-proposals/security-model.md

```diff
@@ -0,0 +1,190 @@
+# Requirements
+
+This document considers "customer facing" entities. For our purposes these are
+the Kubernetes API, and running workloads.
+
+1. Kubernetes needs to be safe in a multi-tenant environment
+2. Securing Kubernetes should be easy
+
+This document uses the "[CIA triad](https://en.wikipedia.org/wiki/Information_security#Key_concepts)" for analyzing these goals.
+
+# Kubernetes API access controls
+
+Kubernetes needs access controls capable of protecting the CIA of cluster
+resources in a multi-tenant environment.
+
+## Concerns with method based access controls
+
+Many Kubernetes deployments provide method based access control (e.g.,
+list pods). This is akin to securing \*nix with permissions on the "open"
+syscall. To allow more than privileged user, ACLs should be placed on nouns, not
+verbs.
+
+## Concerns with implicit relationships
+
+Kubernetes' implicit object relationships create subtle dangers.
+
+1. The "create pod" privilege allows a user to mount a secret, even if the user
```

Comment:
You're right: protecting k8s' objects is insufficient to protect their integrity. The relationships between objects must also be protected (e.g., services <> pods, deployment <> pods, pods <> secrets).

Scoping permissions to a namespace protects most inter-object relationships, because most are bound to a namespace. There are some difficulties with this approach

1. Objects not bound to a namespace

1. The loss of versatility could make separating privileges hard

1. The cluster administrator is responsible for keeping namespaces in sync with ownership

The advantages with namespace are that it can be done today with RBAC.